homeassistant/components/beewi_smartclim/sensor.py

```diff
@@ -0,0 +1,187 @@
+"""Platform for beewi_smartclim integration."""
+import logging
+
+from datetime import datetime, timedelta
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_MONITORED_CONDITIONS,
+    CONF_MAC,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_TEMPERATURE,
+    DEVICE_CLASS_BATTERY,
+)
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+# Default values
+DEFAULT_NAME = "BeeWi SmartClim"
+
+# Sensor config
+SENSOR_TYPES = {
+    "temperature": [DEVICE_CLASS_TEMPERATURE, "Temperature", TEMP_CELSIUS],
+    "humidity": [DEVICE_CLASS_HUMIDITY, "Humidity", "%"],
+    "battery": [DEVICE_CLASS_BATTERY, "Battery", "%"],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_MAC): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_MONITORED_CONDITIONS, default=list(SENSOR_TYPES)): vol.All(
+            cv.ensure_list, [vol.In(SENSOR_TYPES)]
+        ),
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the beewi_smartclim platform."""
+    _LOGGER.info("BeeWi SmartClim has been loaded")
+
+    mac = config.get(CONF_MAC)
+
+    poller = BeewiSmartClimPoller(mac)
+
+    sensors = []
+
+    for parameter in config[CONF_MONITORED_CONDITIONS]:
+        device = SENSOR_TYPES[parameter][0]
+        name = SENSOR_TYPES[parameter][1]
+        unit = SENSOR_TYPES[parameter][2]
+
+        prefix = config.get(CONF_NAME)
+        if prefix:
+            name = "{} {}".format(prefix, name)
+
+        sensors.append(BeewiSmartclimSensor(poller, name, mac, device, unit))
+
+    add_entities(sensors)
+
+
+class BeewiSmartclimSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, poller, name, mac, device, unit):
+        """Initialize the sensor."""
+        self._poller = poller
+        self._name = name
+        self._mac = mac
+        self._device = device
+        self._unit = unit
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor. Convert Celsius to Farhenheit if needed."""
+        if self._device == DEVICE_CLASS_TEMPERATURE and self._unit == TEMP_FAHRENHEIT:
+            return (self._state * 1.8) + 32
```

Comment:
Temperature state is converted by the base entity class. We can remove this case.